### PR TITLE
RedCode: Remove legacy GFXFramework and legacy materials

### PR DESCRIPTION
**Pull request by AMZN-alexpete**
Signed-off-by: Alex Peterson <26804013+AMZN-alexpete@users.noreply.github.com>

## What does this PR do?

Removes 
- legacy GFXFramework
- legacy Materials, .mtl files and related code in TrackView and SceneAPI
- legacy .ent files
- legacy .lua files
- legacy unused .xml files
- legacy unused editor .png files
- legacy particle editor files
- legacy materialFX files
- legacy level files
- misc AutomatedTesting tests that used .mtl and .ent files
- unused CLightAnimWrapper singleton

## How was this PR tested?

- Compiled, ran editor with AutomatedTesting project 
- created a TrackView Sequence
- opened UI editor 
- checked logs for missing file errors

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]
Thanks for doing this!

**Comment by AMZN-alexpete** [comment]
Looks like I need to remove the legacy `LevelBuilderTest` which is causing LmbrCentral AR failures

**Review by amzn-tommy — APPROVED** [review]
nit: rather than just rename Mannequin.svg->SimpleMotion.svg in the engine's Assets folder, I would actually move that into the Assets folder of the EMotionFX Gem.

Also, it looks like .xmlschema is an asset type in AzCore. You should ping sig-core to see if that can be red-coded entirely.

**Review by amzn-rhhong — APPROVED** [review]
LGTM! Great job Alex

**Review by lsemp3d — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/assetpipeline/asset_processor_tests/asset_processor_batch_dependency_tests.py [hunk 1]
```diff
@@ -49,20 +49,12 @@ class TestsAssetProcessorBatch_DependenycyTests(object):
     AssetProcessorBatch Dependency tests
     """
     schemas = [
-        ("C16877167", ".ent"),
-        ("C16877168", "Environment.xml"),
-        ("C16877169", "flares%.xml"),
         ("C16877170", "fonts%.xml"),
-        ("C16877171", "game.xml"),
-        ("C16877172", "particles%.xml"),
+        ("C16877171", "game.xml")
```
Comment (lumberyard-employee-dm) [inline AutomatedTesting/Gem/PythonTests/assetpipeline/asset_processor_tests/asset_processor_batch_dependency_tests.py:53]: The game.xml doesn't exist anymore either.

**Review by lumberyard-employee-dm — COMMENTED** [review]
We can also update the [AssetProcessorPlatformGameConfig.setreg](https://github.com/o3de/o3de/blob/development/AutomatedTesting/AssetProcessorGamePlatformConfig.setreg#L5-L7) in the AutomatedTesting project to remove the setting to ignore processing .cgf files.

**Review by lumberyard-employee-dm — APPROVED** [review]

**Comment by lumberyard-employee-dm** [comment]
> nit: rather than just rename Mannequin.svg->SimpleMotion.svg in the engine's Assets folder, I would actually move that into the Assets folder of the EMotionFX Gem.
> 
> Also, it looks like .xmlschema is an asset type in AzCore. You should ping sig-core to see if that can be red-coded entirely.

It looks like the [CopyDependencyBuilderTest](https://github.com/o3de/o3de/blob/5b85a35f55c45ac0ce998beeda5c7b737e178478/Gems/LmbrCentral/Code/Tests/Builders/CopyDependencyBuilderTest.cpp#L91) is the only thing using the xmlschema asset.
@AMZN-stankowi would know whether we still need it.

**Comment by AMZN-alexpete** [comment]
Looks like LyShine sprites were using the `LmbrCentral::TextureAsset` - I'll move that into the LyShine Gem

**Review by moraaar — APPROVED** [review]
This is great! So much red code! Thanks

File: Assets/Engine/Schema/enginedependency.xmlschema [hunk 2]
```diff
@@ -1,38 +0,0 @@
-<ObjectStream version="3">
```
Comment (AMZN-stankowi) [inline Assets/Engine/Schema/enginedependency.xmlschema:1]: Why are you deleting this xml schema file? Isn't the _dependencies.xml feature still available? This should probably be kept.

File: AutomatedTesting/Gem/PythonTests/assetpipeline/asset_processor_tests/asset_processor_batch_dependency_tests.py [hunk 3]
```diff
@@ -49,20 +49,12 @@ class TestsAssetProcessorBatch_DependenycyTests(object):
     AssetProcessorBatch Dependency tests
     """
     schemas = [
-        ("C16877167", ".ent"),
```
Comment (AMZN-stankowi) [inline AutomatedTesting/Gem/PythonTests/assetpipeline/asset_processor_tests/asset_processor_batch_dependency_tests.py:52]: All that's left in this test is the font.xml scan, right? That should be fine.

File: Gems/LmbrCentral/Code/Source/Builders/CopyDependencyBuilder/XmlBuilderWorker/XmlBuilderWorker.cpp [hunk 4]
```diff
@@ -353,7 +353,6 @@ namespace CopyDependencyBuilder
         AssetBuilderSDK::AssetBuilderDesc xmlSchemaBuilderDescriptor;
         xmlSchemaBuilderDescriptor.m_name = "XmlBuilderWorker";
         xmlSchemaBuilderDescriptor.m_patterns.push_back(AssetBuilderSDK::AssetBuilderPattern("(?!.*libs\\/gameaudio\\/).*\\.xml", AssetBuilderSDK::AssetBuilderPattern::PatternType::Regex));
-        xmlSchemaBuilderDescriptor.m_patterns.push_back(AssetBuilderSDK::AssetBuilderPattern("*.ent", AssetBuilderSDK::AssetBuilderPattern::PatternType::Wildcard));
         xmlSchemaBuilderDescriptor.m_patterns.push_back(AssetBuilderSDK::AssetBuilderPattern("*.vegdescriptorlist", AssetBuilderSDK::AssetBuilderPattern::PatternType::Wildcard));
         xmlSchemaBuilderDescriptor.m_busId = azrtti_typeid<XmlBuilderWorker>();
         xmlSchemaBuilderDescriptor.m_version = 8;
```
Comment (AMZN-stankowi) [inline Gems/LmbrCentral/Code/Source/Builders/CopyDependencyBuilder/XmlBuilderWorker/XmlBuilderWorker.cpp:358]: You should bump the version of the builder up when making a change to it.

**Review by AMZN-stankowi — CHANGES_REQUESTED** [review]
Thanks for doing this. I've got two requested changes before this goes in, the version of the builder should be bumped and the _dependencies.xml schema shouldn't be removed.

File: Assets/Engine/Schema/enginedependency.xmlschema (hunk 2 above)
Comment (AMZN-alexpete) [reply Assets/Engine/Schema/enginedependency.xmlschema:1]: good catch!

**Review by amzn-mike — APPROVED** [review]

**Review by AMZN-stankowi — APPROVED** [review]

**Review by michabr — APPROVED** [review]
LyShine part looks good!

**Pull request merged**